bool state
int8 torque